src/include/migraphx/op/select_module.hpp

```diff
@@ -72,21 +96,48 @@ struct select_module
         {
             MIGRAPHX_THROW("SELECT_MODULE: no compatible submodules found for given input shapes");
         }
+
         auto* module_to_run = *module_iter;
-        std::unordered_map<std::string, argument> params;
+        std::unordered_map<std::string, argument> p_map;
 
-        // add input parameters
-        auto param_names = module_to_run->get_parameter_names();
-        assert(param_names.size() <= args.size());
-        std::transform(param_names.begin(),
-                       param_names.end(),
+        // add input parameters to parameter_map
+        auto in_param_names = get_input_parameter_names(module_to_run);
+        assert(in_param_names.size() <= args.size());
+        std::transform(in_param_names.begin(),
+                       in_param_names.end(),
                        args.begin(),
-                       std::inserter(params, params.end()),
-                       [](auto&& name, auto&& a) { return std::make_pair(name, a); });
+                       std::inserter(p_map, p_map.end()),
+                       [&](auto&& name, auto&& a) { return std::make_pair(name, a); });
 
-        auto results = run(module_to_run, params);
+        // One tuple output parameter in main module to multiple output parameters in submodule
+        auto out_param_names    = get_output_parameter_names(module_to_run);
+        auto output_sub_objects = args.back().get_sub_objects();
+        std::transform(out_param_names.begin(),
+                       out_param_names.end(),
+                       output_sub_objects.begin(),
+                       std::inserter(p_map, p_map.end()),
+                       [&](auto&& name, auto&& a) {
+                           auto ps = module_to_run->get_parameter_shape(name);
+                           decltype(std::make_pair(name, a)) ret;
+                           if(a.get_shape() != ps)
```

Comment:
I thought `compute-shape` would have a tuple of shapes. so it would match exactly. 

Why wouldn't it match?

Reply:
I believe this is for the output. The output will return a dynamic shape, but after running the module we will have a static shape.

Reply:
The shape might not match because of purely shape changes after a gpu instruction in the submodule. For example, when I do a reduce into a squeeze the output parameter the reduce has shape like [4, 1, 2] but the output shape is [4, 2] after the reduce.

Reply:
Do we have test coverage for this case ?

Reply:
Yes, it's the previous `test_select_module_reduce` verify test.